### ISSUE TITLE
Indent on wrap

### ISSUE THREAD

**Issue by Unisay** [rikvdkleij/intellij-haskell#187]
It would be awesome if wrapping behavior respects the scope by making an appropriate indentation:

Let me demonstrate an actual behavior:

![wherewrap](https://cloud.githubusercontent.com/assets/1009751/26443141/8420ab4e-4137-11e7-90e5-5593cac93382.gif)

And the desired behavior is that upon wrapping line its auto-indented.

(If its hard to do in general, then it could still be useful after "where" as demonstrated)

**Comment by rikvdkleij** [comment]
Yes, I can implement solution for your specific example. I will use the number of indentation spaces from the project style settings.

Your example is clear but for example:
```
foo = 
  do 
    bar 
  where
    bar = return 1
```
In this case I have to use as starting point for indentation the column number of `w` in `where`.

I will have to investigate how to implement a general solution because formatting is a matter of taste (personal style) and it also has to produce proper formatted Haskell code. Also the `enter` can be typed before the keyword. General solution should also not be annoying.
